datastore/common/common.go

```diff
@@ -1,11 +1,118 @@
+	// 'name', with a default value of 'defaultVal', and with the option of
+	// expressing the value as an ISO-8601 datetime as a fallback.
+	getSecs := func(name string, defaultVal int64, allowIso8601 bool) int64 {
+		val0 := strings.ToLower(Getenv(name, fmt.Sprintf("%d", defaultVal)))
+		var val int64
+		var err error
+
+		// first attempt to extract directly from int64
+		if val, err = strconv.ParseInt(val0, 10, 64); err != nil {
+			if allowIso8601 {
+				// then attempt to extract from ISO-8601 form
+				var t time.Time
+				if t, err = iso8601ToTime(val0); err != nil {
+					log.Printf(
+						"WARNING: failed to parse %s as an ISO-8601 datetime of the form "+
+						"2023-10-10T00:00:00Z: %s; falling back to default secs: %d",
+						name, val0, defaultVal)
+					val = defaultVal
+				} else {
+					val = t.Unix()
+				}
+			} else {
+				log.Printf(
+					"WARNING: failed to parse %s as an int64: %s; falling back to default secs: %d",
+					name, val0, defaultVal)
+				val = defaultVal
+			}
+		}
+		return val
+	}
+
+	loTimeName := "LOTIME"
+	hiTimeName := "HITIME"
+	defaultLoTimeSecs := int64(86400)
+	defaultHiTimeSecs := int64(0)
+
+	if dynamicTime {
+		loTimeSecs = getSecs(loTimeName, defaultLoTimeSecs, false)
+		hiTimeSecs = getSecs(hiTimeName, defaultHiTimeSecs, false)
+	} else {
+		// NOTE: in this case the defaults make little sense as they define a negative interval,
+		// but are kept like this to keep the documentation (e.g. in README) simple.
+		// A negative interval will be caught and fixed below. In any case, it hardly makes any
+		// sense to use defaults at all in the static case (i.e. you are likely to always want to
+		// specify the valid time range explicitly to match your test setup).
+		loTimeSecs = getSecs(loTimeName, defaultLoTimeSecs, true)
+		hiTimeSecs = getSecs(hiTimeName, defaultHiTimeSecs, true)
+	}
+
+	if hiTimeSecs <= loTimeSecs {
+		log.Printf(
+			"WARNING: hiTimeSecs (%d) <= loTimeSecs (%d); setting hiTimeSecs to loTimeSecs + 1",
+			hiTimeSecs, loTimeSecs)
+		hiTimeSecs = loTimeSecs + 1
+	}
+}
+
+func init() { // automatically called once on program startup (on first import of this package)
+	initValidTimeRange()
+}
+
+// See https://www.pauladamsmith.com/blog/2011/05/go_time.html
+var iso8601layout = "2006-01-02t15:04:05z" // note lower case!
+
+// iso8601ToTime converts time string ts of the form YYYY-MM-DDThh:mm:ssZ to time.Time.
+// Returns (val, nil) upon success, otherwise (time.Time{}, error).
+func iso8601ToTime(ts string) (time.Time, error) {
+        tm, err := time.Parse(iso8601layout, strings.ToLower(ts))
+        if err != nil {
+                return time.Time{}, fmt.Errorf("time.Parse() failed for %s: %v", ts, err)
+        }
+        return tm, nil
+}
+
 // Getenv returns the value of an environment variable or a default value if
 // no such environment variable has been set.
 func Getenv(key string, defaultValue string) string {
```

Comment:
Struggling with github....

I think this is a bit cleaner in this case:
```
	value, ok := os.LookupEnv(key)
	if !ok {
		value = defaultValue
	}
```

Reply:
agreed, will fix!